crates/ra_hir_def/src/nameres.rs

```diff
@@ -149,6 +149,16 @@ static BUILTIN_SCOPE: Lazy<FxHashMap<Name, Resolution>> = Lazy::new(|| {
         .collect()
 });
 
+/// Shadow mode for builtin type
+/// Builtin type can be shadowed by same name mode
```

Comment:
this should say 'module', I guess :slightly_smiling_face: (also, I think the two lines might be squished together in the doc output, the sentences should probably end with `.`)